moveit_ros/hybrid_planning/local_planner/trajectory_operator_plugins/src/simple_sampler.cpp

```diff
@@ -68,11 +68,19 @@ SimpleSampler::addTrajectorySegment(const robot_trajectory::RobotTrajectory& new
   // Reset trajectory operator to delete old reference trajectory
   reset();
 
+  std::cout << "Num waypoints in new_trajectory: " << new_trajectory.getWayPointCount() << std::endl;
+
   // Throw away old reference trajectory and use trajectory update
   reference_trajectory_ = std::make_shared<robot_trajectory::RobotTrajectory>(new_trajectory);
 
-  // Parametrize trajectory and calculate velocity and accelerations
-  time_parametrization_.computeTimeStamps(*reference_trajectory_);
+  // If there are very few waypoitns, time-parameterize the trajectory (calculate velocity and accelerations)
+  if (reference_trajectory_->getWayPointCount() <= 2 && !time_parametrization_.computeTimeStamps(*reference_trajectory_))
+  {
+    std::cout << "Time parameterization failed!" << std::endl;
+    feedback_.feedback = "Time parameterization failed.";
```

Comment:
@sjahr Do you have any suggestion what to do here if time parameterization fails? Filling the feedback message is nice but that won't stop the local planner from trying to execute, will it?

Reply:
@AndyZe You're right, I think it is not possible right now to stop the local planner at this point except killing the node and thus failing the whole hybrid planning process. I've created an [issue](https://github.com/ros-planning/moveit2/issues/964) with a rough idea of how to add this. What do you think about it?